### DISCUSSION TITLE
stop connection lines from merging

### DISCUSSION THREAD

**Discussion by simpsoneric** [terrastruct/TALA#89] [Q&A]
Using the tala layout engine, I have a diagram where I need to keep the connections independent.

https://play.d2lang.com/?script=SsvP50pKLOLiyk3MzOPiSsvPV9DVVQBxrBTyEnNTU0CyCKHi_NzUkozMvHQuQAAAAP__&layout=tala&

<img width="421" alt="image" src="https://github.com/user-attachments/assets/1bce325f-d2ac-4495-ad3d-e1250dd2b160">

This is a simple example above, but I have a shared resource with many different physical connections.  Is there a way to force tala to keep connections from merging?

This is somewhat related to the ideas in:

- https://github.com/terrastruct/d2/discussions/605

**Comment by alixander** [comment]
Yeah I think ports will be how this is done, but support for ports is still a way's out unfortunately.

**Comment by simpsoneric** [comment]
While I don't think my approach below is a "real" solution, in small parts I was able to force a layout like:

https://play.d2lang.com/?script=nJHNaoUwEIX3eYrh3l3hiv3bhFLoA3TVBygxGTU0TiQZF5fiu5f4E_TabpqFkG9ODnOO5_0RZ3jTPCgH2hMjMbQYUNyqLPUDx0LB5RU6ZalQK6oyqlakM9LCDzwxAy-XmZnMMDPMrM6sFuK46wdfnaXmMBHaqRgxSvgWAACtNQbpM2VSljCsPJ3IV4dFbZ2TwEFR7FVA4pt55OC_8KgYp2-HoUGzdWXfSyjz1WHNu7uq0Ek4nRaTUYiUUsK7spS6J9Tsw2I4pQF5iCHGpeOt7hdZmjXBmov2bugoSrgX49rxv95Ov-Ru_3QuYe4bbdOyhMeyTNF2S26amVop1_B_zOFhNtmve1A9PSfVTwAAAP__&layout=tala&

<img width="489" alt="image" src="https://github.com/user-attachments/assets/8d097022-5fa6-4b10-b5e6-f85179a77454">

The thought was:

- have independent nodes within the container
- make those nodes transparent
- stack those nodes all on-top of each other

Still a rough WIP, because there is likely all sorts of ways this will fail.
